ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/ethernet_interfaces.py

```diff
@@ -130,7 +135,13 @@ def ethernet_interfaces(self) -> dict | None:
                             # Strip None values from vlan before adding to list
                             interface = {key: value for key, value in interface.items() if value is not None}
 
-                            ethernet_interfaces[interface_name] = interface
+                            if interface_name in interface_names:
+                                for idx, eth_int in enumerate(ethernet_interfaces):
```

Comment:
Same here and below for subinterfaces, raise an error instead of just overwriting.